shenyu-plugin/shenyu-plugin-proxy/shenyu-plugin-springcloud/src/main/java/org/apache/shenyu/plugin/springcloud/loadbalance/ShenyuSpringCloudServiceChooser.java

```diff
@@ -45,10 +48,10 @@ public final class ShenyuSpringCloudServiceChooser {
     
     private static final Logger LOG = LoggerFactory.getLogger(ShenyuSpringCloudServiceChooser.class);
 
-    private final DiscoveryClient discoveryClient;
+    private ShenyuInstanceRegisterRepository repository;
 
-    public ShenyuSpringCloudServiceChooser(final DiscoveryClient discoveryClient) {
-        this.discoveryClient = discoveryClient;
+    public ShenyuSpringCloudServiceChooser(final RegisterConfig config) {
+        this.repository = ShenyuInstanceRegisterRepositoryFactory.newAndInitInstance(config);
```

Comment:
It should not be initialized here. You need to consider getting the latest instance in the method after modification.